File: app/src/main/java/com/boardgamegeek/sorter/CurrentValueSorter.java
package com.boardgamegeek.sorter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import com.boardgamegeek.R;
import com.boardgamegeek.provider.BggContract.Collection;

public class CurrentValueSorter extends MoneySorter {
	public CurrentValueSorter(Context context) {
		super(context);
	}

	@StringRes
	@Override
	protected int getDescriptionId() {
		return R.string.collection_sort_current_value;
	}

	@StringRes
	@Override
	protected int getTypeResource() {
		return R.string.collection_sort_type_current_value;
	}

	@NonNull
	@Override
	protected String getAmountColumnName() {
		return Collection.PRIVATE_INFO_CURRENT_VALUE;
	}

	@NonNull
	@Override
	protected String getCurrencyColumnName() {
		return Collection.PRIVATE_INFO_CURRENT_VALUE_CURRENCY;
	}
}

File: app/src/main/java/com/boardgamegeek/sorter/CurrentValueSorter.kt
package com.boardgamegeek.sorter

import android.content.Context
import androidx.annotation.StringRes
import com.boardgamegeek.R
import com.boardgamegeek.entities.CollectionItem

class CurrentValueSorter(context: Context) : MoneySorter(context) {
    override val ascendingSortTypeResId: Int
        @StringRes
        get() = R.string.collection_sort_type_current_value_asc

    override val descendingSortTypeResId: Int
        @StringRes
        get() = R.string.collection_sort_type_current_value

    override val descriptionResId: Int
        @StringRes
        get() = R.string.collection_sort_current_value

    override fun amount(item: CollectionItem) = item.currentValue

    override fun currency(item: CollectionItem) = item.currentValueCurrency
}
